### PR TITLE
Bug 1982778: Favour HTTP probes over exec for Thanos query

### PR DESCRIPTION
**Pull request by PhilipGough**
This is cherry pick of #1277 which backports to 4.8.z

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: jsonnet/node-exporter.libsonnet [hunk 1]
```diff
@@ -138,7 +138,19 @@ function(params)
...
+                            [
+                              '--collector.cpu.info',
+                              '--collector.textfile.directory=' + textfileDir,
+
+                              // The cpufreq collector seems to be causing high
+                              // load on some nodes with lots of cores.  Disable
+                              // it temporarily as a workaround.
+                              //
+                              // https://bugzilla.redhat.com/show_bug.cgi?id=1972076
+                              // https://github.com/prometheus/node_exporter/issues/1880
```
Comment (prashbnair) [inline jsonnet/node-exporter.libsonnet:151]: Why do we want to do this? It does not seem related to the bug reported

Comment (PhilipGough) [reply jsonnet/node-exporter.libsonnet:151]: correct, that shouldnt be there :)
